### ISSUE TITLE
[Bug]: juju.errors.JujuAPIError: client incompatible with server 3.1-beta1

### ISSUE THREAD

**Issue by Abuelodelanada** [juju/python-libjuju#773]
### Description

Integration tests failed with `juju.errors.JujuAPIError`

```
Traceback (most recent call last):
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py", line 301, in _asyncgen_fixture_wrapper
    result = event_loop.run_until_complete(setup())
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py", line 283, in setup
    res = await gen_obj.__anext__()
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_operator/plugin.py", line 231, in ops_test
    await ops_test._setup_model()
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_operator/plugin.py", line 686, in _setup_model
    await self._controller.connect(self.controller_name)
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/controller.py", line 136, in connect
    await self.update_endpoints()
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/controller.py", line 139, in update_endpoints
    info = await self.info()
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/controller.py", line 283, in info
    return await controller_facade.ControllerAPIInfoForModels(entities=params)
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/client/facade.py", line 481, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/client/_client11.py", line 650, in ControllerAPIInfoForModels
    reply = await self.rpc(msg)
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/client/facade.py", line 654, in rpc
    result = await self.connection.rpc(msg, encoder=TypeEncoder)
  File "/home/ubuntu/repos/loki-k8s-operator/.tox/integration/lib/python3.10/site-packages/juju/client/connection.py", line 638, in rpc
    raise errors.JujuAPIError(result)
juju.errors.JujuAPIError: client incompatible with server 3.1-beta1
```

### Urgency

Annoying bug in our test suite

### Python-libjuju version

3.0.4

### Juju version

3.1-beta1

### Reproduce / Test

```python
git clone https://github.com/canonical/loki-k8s-operator
tox -e integration -- -k test_alert_rules_fire.py
```

**Comment by Abuelodelanada** [comment]
```
$ cat ~/.local/share/juju/controllers.yaml | yq '.controllers.charm-dev-batteries.agent-version'
3.1-beta1
```

**Comment by juanmanuel-tirado** [comment]
Compatibility with the next Juju 3.1.x will not be available until we release python-libjuju 3.1.0. We will have to wait :)
Thanks for reporting.

**Comment by juanmanuel-tirado** [comment]
Hola,

No it's not there yet. We are going to follow a conservative approach when supporting juju APIs that are not release candidates. The main reason is that to bring some stability to python-libjuju we have to wait for a stable version of the juju API. Said that, you may check the `master` branch from time to time. No python-libjuju supporting juju 3.1 will be released until juju 3.1 is in candidate.

And you're right this is not really a `done` but I hope you understand the reasons for closing the bug :) And gracias for reporting!
